packages/ra-core/src/types.ts

```diff
@@ -141,6 +141,11 @@ export interface GetListResult<RecordType extends RaRecord = any> {
     };
 }
 
+export interface GetInfiniteListResult<RecordType extends RaRecord = any> {
+    data: RecordType[];
+    total?: number;
+    pageParam?: number;
+}
```

Comment:
I think you are missing

```ts
    pageInfo?: {
        hasNextPage?: boolean;
        hasPreviousPage?: boolean;
    };
```

because you are using them in the logic of your `getNextPageParam` and `getPreviousPageParam` functions in `useInfiniteGetList.ts`